MathLib/LinAlg/PETSc/PETScVector.cpp

```diff
@@ -217,18 +212,34 @@ void PETScVector::viewer(const std::string &file_name, const PetscViewerFormat v
     PetscViewerASCIIOpen(PETSC_COMM_WORLD, file_name.c_str(), &viewer);
     PetscViewerPushFormat(viewer, vw_format);
 
-    PetscObjectSetName((PetscObject)_v, file_name.c_str());
-    VecView(_v, viewer);
+    PetscObjectSetName((PetscObject)*_v, file_name.c_str());
+    VecView(*_v, viewer);
 
 #define  nEXIT_TEST
 #ifdef EXIT_TEST
-    VecDestroy(&_v);
+    VecDestroy(_v);
     PetscFinalize();
     exit(0);
 #endif
 
 }
 
+void PETScVector::shallowCopy(const PETScVector &v)
+{
+    destroy();
+
+    _v.reset(new PETSc_Vec);
+
+    VecDuplicate(*v._v, _v.get());
+
+    // TODO can't that be copied from v?
```

Comment:
VecDuplicate is called, then the storage structures of two vectors are the same. The following variables can be assigned to the new vector directly.